source/Calamari.Shared/Integration/Packages/Download/FeedLoginDetailsProvider.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using Calamari.CloudAccounts;
+using Calamari.Common.Plumbing.Variables;
+
+namespace Calamari.Integration.Packages.Download
+{
+    public interface IFeedLoginDetailsProvider
+    {
+        (string Username, string Password, string FeedUri) GetFeedLoginDetails(
+            IVariables variables,
+            string username,
+            string password);
+    }
+
+    public class FeedLoginDetailsProvider : IFeedLoginDetailsProvider
```

Comment:
Since this only supports AWS ECR at this stage can we rename it to something ECR specific? If we want to support others and make the implementation generic in the future we can but for now I think being explicit here makes sense.